### PR TITLE
RHMAP-18011 Provision OSX review

### PR DESCRIPTION
**Pull request by pmacko1**

### CODE REVIEW

**Comment by pmacko1** [comment]
my mistake

File: provision-osx/README.md [hunk 1]
```diff
@@ -12,151 +12,145 @@ Provision Jenkins node for the Digger. This role performs the following tasks:
...
-<aside class="warning">
-Only proxies that do not require authentication are supported.
-</aside>
+* `xcode_install_user` and `xcode_install_password` variables set, otherwise Xcode tasks will be skipped.
 
+## Required Network Access
 
 The node will need access to the following external hosts:
-* `https://github.com` - To download and install github and Cocoapods.
+* `https://github.com` - To download and install github(???) and Cocoapods.
```
Comment (pmacko1) [inline provision-osx/README.md]: github application (https://desktop.github.com/) ?

File: provision-osx/README.md [hunk 2]
```diff
@@ -12,151 +12,145 @@ Provision Jenkins node for the Digger. This role performs the following tasks:
...
 
-To run this section as a standalone step you must specify the `osx_install_homebrew` tag.
 
-### Options
+### Variables
 * `homebrew_version` - The version of Homebrew to install (git tag).
 * `homebrew_packages` - The packages to install using Homebrew. Format: `{ name: <PACKAGE_NAME> }`.
 * `homebrew_repo` - The git repo where Homebrew resides (defaults to GitHub repo).
-* `homebrew_prefix` -
+* `homebrew_prefix` - (???)
```
Comment (pmacko1) [inline provision-osx/README.md]: what is this?

File: provision-osx/README.md [hunk 3]
```diff
@@ -12,151 +12,145 @@ Provision Jenkins node for the Digger. This role performs the following tasks:
...
-To run this section as a standalone step you must specify the `osx_install_nodejs` tag.
+Installs NVM along with versions of Node specified in variables.
+Packages from NPM can be specified for installation too, but they will be installed globally.
 
-### Options
+
+### Variables
 * `nvm_install_url` - The URL of NVM installation script (defaults to GitHub release).
-* `nvm_install_file_name` - What to name the file on the node.
+* `nvm_install_file_name` - What to name the file on the node. (???)
```
Comment (pmacko1) [inline provision-osx/README.md]: installation filename?

File: provision-osx/README.md [hunk 4]
```diff
@@ -12,151 +12,145 @@ Provision Jenkins node for the Digger. This role performs the following tasks:
...
-### Options
-* `credential_private_key` - Private key stored in Jenkins and used to SSH into the macOS node. If this is not set then a key pair will be generated.
-* `credential_public_key` - Public key of the pair. If this is not set then a key pair will be generated.
-* `credential_passphrase` - Passphrase of the private key. This is stored in Jenkins and used to SSH into the macOS node. If this is not set the private key will not be password protected.
-* `buildfarm_node_port` - The port to connect to the macOS node on. Defaults to `22`.
+### Variables
+* `credential_private_key` - Private key stored in Jenkins and used to SSH into the macOS node. If not set, then asymmetric key pair will be generated.
+* `credential_public_key` - Public key from asymmetric key pair. If this is not set then a key pair will be generated.
+* `credential_passphrase` - Password for the encryption of the private key. This is stored in Jenkins and used to SSH into the macOS node. If this is not set the private key will not be encrypted.
+* `buildfarm_node_port` - The port to connect to the macOS node to. Defaults to `22`.
```
Comment (pmacko1) [inline provision-osx/README.md]: remove middle 'to'

File: provision-osx/README.md [hunk 5]
```diff
@@ -12,151 +12,145 @@ Provision Jenkins node for the Digger. This role performs the following tasks:
...
+
+Configures a proxy to be used within the macOS node.
+
+***CAUTION: Only proxies that do not require authentication are supported by the Ansible job.***
 
-Configures a proxy to be used whithin the osx node.
 
-To run this section as a standalone step you must specify the `osx_configure_proxy` tag.
 
 ### Options
```
Comment (pmacko1) [inline provision-osx/README.md]: Variables

File: provision-osx/README.md (hunk 1 above)
Comment (matskiv) [reply provision-osx/README.md]: GitHub repositories

File: provision-osx/README.md (hunk 2 above)
Comment (matskiv) [reply provision-osx/README.md]: Its a prefix used for homebrew_install_path (next line), not sure how to describe it.

File: provision-osx/README.md (hunk 3 above)
Comment (matskiv) [reply provision-osx/README.md]: Name of NVM installation script.

File: provision-osx/README.md [hunk 6]
```diff
@@ -12,144 +12,145 @@ Provision Jenkins node for the build farm. This role performs a few tasks, these
...
 * `npm_packages` - A list of global NPM packages to install. Format: `{ name: <PACKAGE_NAME>, version: <PACKAGE_VERSION> }`.
 
-## Install Xcode
-Installs Xcode CLI tools and Xcode versions specified.
-Installing Xcode can take some time (~ 30 minutes without a cached copy) so please
-be patient during these stages.
+## Install Xcode and Xcode CLI tools
+To run this section as a standalone step use `osx_install_xcode` tag in the playbook.
+
+***Installing Xcode can take ~ 30 minutes without a cached copy.***
```
Comment (matskiv) [inline provision-osx/README.md]: btw: It's 30 minutes **per installed version**, maybe we should add this piece of information while we are at it..

Comment (odra) [reply provision-osx/README.md]: I have experienced random timeouts from apple server as well

File: provision-osx/README.md (hunk 1 above)
Comment (pmacko1) [reply provision-osx/README.md]: resolved

File: provision-osx/README.md (hunk 3 above)
Comment (pmacko1) [reply provision-osx/README.md]: resolved

File: provision-osx/README.md (hunk 4 above)
Comment (pmacko1) [reply provision-osx/README.md]: dones

File: provision-osx/README.md (hunk 5 above)
Comment (pmacko1) [reply provision-osx/README.md]: done

File: provision-osx/README.md (hunk 6 above)
Comment (pmacko1) [reply provision-osx/README.md]: done

File: provision-osx/README.md (hunk 2 above)
Comment (pmacko1) [reply provision-osx/README.md]: hm, i think it is low priority then, ill let it pass

**Pull request merged**